### PR TITLE
refactor: env var handling enhancement

### PR DESCRIPTION
**Pull request by peterdeme**
Env var handling enhancement

### CODE REVIEW

File: lib/stream-chat/client.rb [hunk 1]
```diff
@@ -36,13 +37,15 @@ class Client
...
+    def initialize(api_key, api_secret, timeout = nil, **options)
+      raise ArgumentError, 'api_key and api_secret are required' if api_key.to_s.empty? || api_secret.to_s.empty?
+
+      @api_key = api_key
+      @api_secret = api_secret
+      @timeout = timeout || DEFAULT_TIMEOUT
       @options = options
       @auth_token = JWT.encode({ server: true }, @api_secret, 'HS256')
-      @base_url = options[:base_url] || ENV['STREAM_CHAT_URL'] || BASE_URL
+      @base_url = options[:base_url] || DEFAULT_BASE_URL
```
Comment (ferhatelmas) [inline lib/stream-chat/client.rb:48]: this is breaking if somebody is using from env directly (very unlikely though), let's mention in changelog clearly

File: lib/stream-chat/client.rb [hunk 2]
```diff
@@ -55,9 +58,13 @@ def initialize(api_key = '', api_secret = '', timeout = 6.0, **options)
     end
 
     # initializes a Stream Chat API Client from STREAM_KEY and STREAM_SECRET
-    # environmental variables.
+    # environmental variables. STREAM_CHAT_TIMEOUT and STREAM_CHAT_URL
+    # variables are optional.
     def self.from_env
```
Comment (ffenix113) [inline lib/stream-chat/client.rb:63]: What do you think about something like this(not sure that the code is valid, but should explain the intent):
```ruby
   def self.from_env(**options)
      Client.new(ENV['STREAM_KEY'],
                 ENV['STREAM_SECRET'],
                 ENV['STREAM_CHAT_TIMEOUT'],
                 **({base_url: ENV['STREAM_CHAT_URL']}.merge(options)))
    end
```
?

Comment (peterdeme) [reply lib/stream-chat/client.rb:63]: sure, why not

File: lib/stream-chat/client.rb (hunk 1 above)
Comment (peterdeme) [reply lib/stream-chat/client.rb:48]: yup, this is a pretty new feature so it's unlikely, but I just added it to the body of the commit message. I think the changelog will reflect it. if not, I'll add it manually there.

**Review by ffenix113 — APPROVED** [review]
As always - great work!

**Pull request merged**